ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSemanticAnalyzer.java

```diff
@@ -393,35 +394,22 @@ private void dumpFunctionMetadata(String dbName, Path dumpRoot) throws SemanticE
     }
   }
 
-  /**
-   *
-   * @param ast
-   * @param dbName
-   * @param tblName
-   * @param dbRoot
-   * @return tbl dumped path
-   * @throws SemanticException
-   */
-  private Path dumpTbl(ASTNode ast, String dbName, String tblName, Path dbRoot) throws SemanticException {
-    Path tableRoot = new Path(dbRoot, tblName);
+  private void dumpTable(ASTNode ast, String dbName, String tblName, Path dbRoot)
+      throws SemanticException {
     try {
-      URI toURI = EximUtil.getValidatedURI(conf, tableRoot.toUri().toString());
       TableSpec ts = new TableSpec(db, conf, dbName + "." + tblName, null);
-
-      ExportSemanticAnalyzer.prepareExport(ast, toURI, ts, getNewReplicationSpec(), db, conf, ctx,
-          rootTasks, inputs, outputs, LOG);
+      TableExport.Paths exportPaths = new TableExport.Paths(ast, dbRoot, tblName, conf);
```

Comment:
Well, there is no problem with creating that object outside. 
I just see TableExport as an object which takes a table specifications and dump directory as inputs and it should take care of making the exportPaths itself. Moreover, we won't have customized implementations of exportPaths creation for tables.
Anyways, I'm not an expert in object oriented programming, and so agree with you... :)